src/contracts/middleware/IndexRegistry.sol

```diff
@@ -0,0 +1,263 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+
+import "../interfaces/IIndexRegistry.sol";
+import "../interfaces/IRegistryCoordinator.sol";
+import "../libraries/BN254.sol";
+
+contract IndexRegistry is IIndexRegistry {
```

Comment:
looks like this contract is missing top-level context / comments -- please add them